lib/medtronic/processData.js

```diff
@@ -426,6 +464,84 @@ function buildBGRecords(records) {
   return postrecords;
 }
 
+function buildCGMRecords(events) {
+
+  var postrecords = [];
+  var start = null;
+  var recordsSinceTimestamp = null;
+
+  // we have to work through the CGM records in reverse, as sensor timestamps
+  // affect previous values and are subject to time changes
+
+  for( var i = events.length - 1 ; i >= 0; i-- ) {
+    var event = events[i];
+
+    if(event.jsDate != false) {
+      console.log('CGM event:',sundial.formatDeviceTime(event.jsDate), event.type.name, common.bytes2hex([event.head]),common.bytes2hex(event.date),common.bytes2hex(event.body),common.bytes2hex(event.descriptor) );
+
+      if(event.type.value === CBG_RECORD_TYPES.SENSOR_TIMESTAMP.value) {
+        start = event.jsDate;
+        recordsSinceTimestamp = 0;
+      }
+
+      if(event.body) {
+        var offset = event.body.length;
+        var index = event.index - offset;
+
+        while(offset >= 0) {
+
+          var isig = struct.extractShort(event.descriptor, offset * 2);
+
+          if(isig !== 0x22 &&  isig > 255) {
+            var cbg = (isig & 0x01) + (event.body[offset] << 1);
+
+            var date = sundial.applyOffset(start, -(recordsSinceTimestamp * 5));
+            recordsSinceTimestamp += 1;
+
+            console.log('CBG:', sundial.formatDeviceTime(date), common.bytes2hex(struct.extractBytes(event.descriptor, offset * 2, 2)),',',common.bytes2hex([struct.extractByte(event.body,offset)]),',',(isig & 0x01) + (event.body[offset] << 1),',',(isig  & 0xFFFE) / 100.0);
+            if(cbg >= 40) { // ignore magic numbers
+              var record = cfg.builder.makeCBG()
+                    .with_value(cbg)
+                    .with_deviceTime(sundial.formatDeviceTime(date))
+                    .with_units('mg/dL')      // TODO: check if mmol/L cbg is also hard-coded in mg/dL
```

Comment:
Not a bad idea, but then we'll have to go back and do it for all the other drivers too? ;) I guess in this specific instance I prefer consistency to prevent confusion, and keep in mind that the data services API will validate the units on ingestion.